apis/v1alpha1/httproute_types.go

```diff
@@ -427,64 +446,114 @@ type HTTPRequestHeaderFilter struct {
-	// that includes the HTTPRoute will be true.
+	// ServiceName refers to the name of the Service to mirror matched requests
+	// to. When specified, this takes the place of BackendRef. If both
+	// BackendRef and ServiceName are specified, ServiceName will be given
+	// precedence. If the referent cannot be found, controllers must set the
+	// "InvalidRoutes" status condition on any Gateway that includes this
+	// Route to true.
 	//
-	// Support: Core (Kubernetes Services)
-	// Support: Implementation-specific (Other resource types)
+	// Support: Core
 	//
-	TargetRef ForwardToTargetObjectReference `json:"targetRef"`
+	// +optional
+	// +kubebuilder:validation:MaxLength=253
+	ServiceName *string `json:"serviceName,omitempty"`
 
-	// TargetPort specifies the destination port number to use for the TargetRef.
-	// If unspecified and TargetRef is a Service object consisting of a single
-	// port definition, that port will be used. If unspecified and TargetRef is
-	// a Service object consisting of multiple port definitions, an error is
-	// surfaced in status.
+	// BackendRef is a local object reference to mirror matched requests to. If
+	// both BackendRef and ServiceName are specified, ServiceName will be given
+	// precedence. If the referent cannot be found, controllers must set the
+	// "InvalidRoutes" status condition on any Gateway that includes this Route
+	// to true.
+	//
+	// Support: Extended
+	//
+	// +optional
+	BackendRef *LocalObjectReference `json:"backendRef,omitempty"`
+
+	// Port specifies the destination port number to use for the backend
+	// referenced by the ServiceName or BackendRef field. If unspecified and a
+	// Service object consisting of a single port definition is the backend,
+	// that port will be used. If unspecified and the backend is a Service
+	// object consisting of multiple port definitions, controllers must set the
+	// "InvalidRoutes" status condition on any Gateway that includes this Route
+	// to true.
 	//
 	// Support: Core
 	//
 	// +optional
-	TargetPort *TargetPort `json:"targetPort,omitempty"`
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=65535
+	Port *int32 `json:"port,omitempty"`
 }
 
-// HTTPForwardingTarget is the target to send the request to for a given a match.
-type HTTPForwardingTarget struct {
-	// To references referenced object(s) where the request should be sent. The
-	// resource may be "services" (omit or use the empty string for the
-	// group), or an implementation may support other resources (for
-	// example, resource "myroutetargets" in group "networking.acme.io").
-	// Omitting or specifying the empty string for both the resource and
-	// group indicates that the resource is "services".  If the referent
-	// cannot be found, the "InvalidRoutes" status condition on any Gateway
-	// that includes the HTTPRoute will be true.
+// HTTPRouteForwardTo defines how a HTTPRoute should forward a request.
+type HTTPRouteForwardTo struct {
```

Comment:
In HTTPRouteForwardTo and RouteForwardTo, 4 fields are common, does it make sense to perform type embedding here?

In HTTPRouteForwardTo, HTTPRequestMirrorFilter, 3 fields are common but this duplication could be okay if we de-duplicate the first case.

Reply:
That a great idea. If it's OK, I think I'd prefer to do that in a follow up just to keep the momentum around this.

Reply:
https://github.com/kubernetes-sigs/service-apis/issues/358 to track.